src/edu/csus/ecs/pc2/core/report/ResultsExportReport.java

```diff
@@ -0,0 +1,279 @@
+// Copyright (C) 1989-2023 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.core.report;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+import edu.csus.ecs.pc2.VersionInfo;
+import edu.csus.ecs.pc2.core.Constants;
+import edu.csus.ecs.pc2.core.IInternalController;
+import edu.csus.ecs.pc2.core.StringUtilities;
+import edu.csus.ecs.pc2.core.Utilities;
+import edu.csus.ecs.pc2.core.execute.ExecuteUtilities;
+import edu.csus.ecs.pc2.core.log.Log;
+import edu.csus.ecs.pc2.core.model.Filter;
+import edu.csus.ecs.pc2.core.model.FinalizeData;
+import edu.csus.ecs.pc2.core.model.IInternalContest;
+import edu.csus.ecs.pc2.exports.ccs.ResultsFile;
+
+/**
+ * Export Results files report.
+ * 
+ * @author Douglas A. Lane <pc2@ecs.csus.edu>
+ *
+ */
+public class ResultsExportReport implements IReport {
+    
+    private IInternalContest contest;
+
+    private IInternalController controller;
+
+    private Log log;
+
+    private Filter filter = new Filter();
+
+    private String pc2ResultsDir = null;
+    
+    private String primaryCCSResultsDir = null;
+
+    /**
+     * 
+     */
+    private static final long serialVersionUID = -796328654541676730L;
+
+    public ResultsExportReport(IInternalContest contest, IInternalController controller, String pc2ResultsDir) {
+        super();
+        this.pc2ResultsDir = pc2ResultsDir;
+        setContestAndController(contest, controller);
+    }
+
+    public ResultsExportReport() {
```

Comment:
I think this constructor is a bad idea. It allows someone to construct a `ResultsExportReport` which has no contest or controller -- potentially resulting in NullPointerExceptions. I think the constructor should be removed, unless a good explanation for it exists (in which case such explanation should be documented in a header comment).